Bottom: 38c4d30ea9146c8e2905ae4a92bd0948fd7ba8cd
Top:    71a37bce3facf1496ea7a10ea6c200c42cca0c05
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:28:24 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
